fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/BaseWhereAdapter.java

```diff
@@ -0,0 +1,1137 @@
+        return getThis();
+    }
+
+    /**
+     * Add a literal double value to the expression
+     * @param value
+     * @return
+     */
+    public T literal(double value) {
+        predicateParser.literal(value);
+        return getThis();
+    }
+
+    // ### EQUAL ###
+    /**
+     * Add an equals '=' node to the expression
+     * @return
+     */
+    public T eq() {
+        predicateParser.eq();
+        return getThis();
+    }
+
+    /**
+     * Add '= <ref>' to the expression
+     * @param ref
+     * @return
+     */
+    public T eq(ColumnRef ref) {
+        predicateParser.eq();
+        predicateParser.column(ref.toString());
+        return getThis();
+    }
+
+    /**
+     * Add '= <literalValue>' to the expression
+     * @param literalValue
+     * @return
+     */
+    public T eq(long literalValue) {
+        predicateParser.eq();
+        predicateParser.literal(literalValue);
+        return getThis();
+    }
+
+    /**
+     * Add '= <literalValue>' to the expression
+     * @param literalValue
+     * @return
+     */
+    public T eq(int literalValue) {
+        predicateParser.eq();
+        predicateParser.literal((long)literalValue);
+        return getThis();
+    }
+
+    /**
+     * Add '= <columnExpression>' to the expression
+     * @param literalValue
+     * @return
+     */
+    public T eq(String columnExpression) {
+        predicateParser.eq();
+        predicateParser.column(columnExpression);
+        return getThis();
+    }
+
+    /**
+     * Add '= <str>' to the expression
+     * @param literalValue
+     * @return
+     */
+    public T eq(LiteralString str) {
+        predicateParser.eq();
+        predicateParser.literal(str.getValue());
+        return getThis();
+    }
+
+    /**
+     * Add '= <tableAlias>.<columnExpression>' to the expression
+     * @param literalValue
+     * @return
+     */
+    public T eq(String tableAlias, String columnExpression) {
+        predicateParser.eq();
+        predicateParser.column(tableAlias, columnExpression);
+        return getThis();
+    }
+
+    /**
+     * Add a bind marker and value to the expression
+     * @param bindMarker
+     * @return
+     */
+    public T eq(BindMarkerNode bindMarker) {
+        predicateParser.eq();
+        predicateParser.bindMarker(bindMarker);
+        return getThis();
+    }
+
+    /**
+     * To simplify code, support multiple where clauses, treating each as an AND
+     * @param columnName
+     * @return
+     */
+    public T where(String columnName) {
+        predicateParser.and();
+        predicateParser.column(columnName);
+        return getThis();
+    }
+
+    /**
+     * To simplify code, support multiple where clauses, treating each as an AND
+     * @param tableAlias
+     * @param columnName
+     * @return
+     */
+    public T where(String tableAlias, String columnName) {
+        predicateParser.and();
+        predicateParser.column(tableAlias, columnName);
+        return getThis();
```

Comment:
and(String tableAlias,String columnName) on 258 is nearly identical.

Reply:
This was added to make it easier to write statements when you don't know if there is already an earlier predicate...but we don't quite support that fully now (because it turned out it wasn't required based on the structure of queries we need to build).